src/server/pps/server/api_server.go

```diff
@@ -75,6 +94,14 @@ func (a *apiServer) CreateJob(ctx context.Context, request *ppsclient.CreateJobR
 		return nil, err
 	}
 
+	// We need to sort job inputs because the following code depends on
+	// the invariant that inputs[i] matches parentInputs[i]
+	sort.Sort(JobInputs(request.Inputs))
```

Comment:
We should probably add some checks here that the parent job have the same set of input repos as the child job. There's technically nothing in the API that forbids this.